### PR TITLE
Added parasail alignment, added fastparquet to dependencies

### PR DESCRIPTION
**Pull request by WojciechowskiDaniel**

### CODE REVIEW

File: modules/local/run_sequence_aligner.nf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env nextflow
```
Comment (twrzes) [inline modules/local/run_sequence_aligner.nf]:
```suggestion
```
This line is not needed to import a module in `main.nf` file

File: modules/local/run_sequence_aligner.nf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    output:
+    path output_path
+
+    script:  // This script is bundled with the pipeline, in nf-core/simseqer/bin/
+    """
+    parasail_alignment.py -query $query_path \
+                     -target $target_path\
+                     -out $output_path
+    REFERENCE_LIST=\$(cat $output_path)
```
Comment (twrzes) [inline modules/local/run_sequence_aligner.nf]: Is this line needed?

File: bin/parasail_alignment.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    similar = counts[':'] + matches
+    gaps = counts[' ']
+    length = len(comp)
+    gap_regex = re.compile(r'\s+')
+    return {
+        'score': result.score,
+        'qseqid': int(query.name),
+        'sseqid': int(target.name),
+        'qstart': cigar.beg_query,
+        'qend': '',
```
Comment (twrzes) [inline bin/parasail_alignment.py:65]: Why is this field empty?

File: bin/parasail_alignment.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        'sstart': cigar.beg_ref,
+        'send': '',
```
Comment (twrzes) [inline bin/parasail_alignment.py:67]: Why is this field empty?

File: bin/parasail_alignment.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        'qlen': len(result.query),
+        'slen': len(result.ref),
+        'pident': matches,
+        'ppos': similar,
+        'similarity': round(float(similar) / length, 2),
+        'identity': round(float(matches) / length, 2),
+        'length': length,
+        'gaps': gaps,
+        'gapopen': len(gap_regex.findall(comp)),
+        'qcov': round(float(length - gaps) / len(query.seq), 2)
```
Comment (twrzes) [inline bin/parasail_alignment.py]: I don't think it is that easy. `gaps` provides counts for both query and reference. To calculate `qcov` you only need to subtract gaps in the query, not in the reference. I am happy to talk about it further :)

File: environment.yml [hunk 6]
```diff
@@ -15,3 +15,4 @@ dependencies:
   - bioconda::blast=2.9.0
   - conda-forge::pandas=1.0.5
   - conda-forge::dask=2.20.0
+  - conda-forge::fastparquet
```
Comment (twrzes) [inline environment.yml]: I think it would be better if you specify a particular version of `fastparquet` library

**Review by twrzes — CHANGES_REQUESTED** [review]

File: modules/local/run_sequence_aligner.nf (hunk 1 above)
Comment (WojciechowskiDaniel) [reply modules/local/run_sequence_aligner.nf]: Removed

File: modules/local/run_sequence_aligner.nf (hunk 2 above)
Comment (WojciechowskiDaniel) [reply modules/local/run_sequence_aligner.nf]: No, removed

File: bin/parasail_alignment.py (hunk 4 above)
Comment (WojciechowskiDaniel) [reply bin/parasail_alignment.py:67]: Calculations for that will be added later, added todo

File: bin/parasail_alignment.py (hunk 3 above)
Comment (WojciechowskiDaniel) [reply bin/parasail_alignment.py:65]: Calculations for that will be added later, added todo

File: environment.yml (hunk 6 above)
Comment (WojciechowskiDaniel) [reply environment.yml]: Added version

**Review by twrzes — APPROVED** [review]

**Pull request merged**